certora/specs/AccountingOracle.spec

```diff
@@ -0,0 +1,867 @@
+    env e; calldataarg args; env e2;
+
+    uint256 consensusVersion; uint256 refSlot;
+    uint256 lastFinalizableWithdrawalRequestId; uint256 simulatedShareRate; bool isBunkerMode;
+    uint256 extraDataFormat; bytes32 extraDataHash; uint256 extraDataItemsCount;
+
+    uint256 contractVersion;
+
+    bytes32 submittedHash = helperCreateAndSubmitReportData( e,
+                                consensusVersion, refSlot,
+                                lastFinalizableWithdrawalRequestId, simulatedShareRate, isBunkerMode,
+                                extraDataFormat, extraDataHash, extraDataItemsCount,
+                                contractVersion );
+    
+    submitReportExtraDataEmpty@withrevert(e2);
+
+    bool submitReverted = lastReverted;
+
+    assert (extraDataFormat != EXTRA_DATA_FORMAT_EMPTY())   => submitReverted;
+    // assert (extraDataItemsCount != 0)                       => submitReverted;  // causes timeout
+    // assert (extraDataHash != 0)                             => submitReverted;  // causes timeout
+}
+
+// 13. Cannot call submitReportExtraDataList() if the report submitted with submitReportData()
+//     had report.extraDataFormat != EXTRA_DATA_FORMAT_LIST()
+// Status: Pass
+// https://vaas-stg.certora.com/output/80942/c1df39d3d1f0469ab6383267518fc0b9/?anonymousKey=2a9b2bc84f79cd8307c7341583361091fb0b3b0a
+rule cannotSubmitReportExtraDataListWhenExtraDataIsEmpty() {
+    require contractAddressesLinked();
+    env e; calldataarg args; env e2;
+
+    uint256 consensusVersion; uint256 refSlot;
+    uint256 lastFinalizableWithdrawalRequestId; uint256 simulatedShareRate; bool isBunkerMode;
+    uint256 extraDataFormat; bytes32 extraDataHash; uint256 extraDataItemsCount;
+
+    uint256 contractVersion;
+
+    bytes32 submittedHash = helperCreateAndSubmitReportData( e,
+                                consensusVersion, refSlot,
+                                lastFinalizableWithdrawalRequestId, simulatedShareRate, isBunkerMode,
+                                extraDataFormat, extraDataHash, extraDataItemsCount,
+                                contractVersion );
+    
+    bytes dataItems;
+    submitReportExtraDataList@withrevert(e2, dataItems);
+
+    bool submitReverted = lastReverted;
+
+    assert (extraDataFormat != EXTRA_DATA_FORMAT_LIST())    => submitReverted;
+    assert (extraDataItemsCount == 0)                       => submitReverted;
+    assert (extraDataHash == 0)                             => submitReverted;
+}
+
+// 14. If the reportExtraDataEmpty() was processed you cannot submit again the same previous submitReportData()
+// Status: Pass
+// https://vaas-stg.certora.com/output/80942/9024662237794b29996f1cccbd33ceb5/?anonymousKey=efb8e3194e19a36b1afee297e8bedf9045d82ee4
+rule cannotSubmitSameReportAfterSubmitExtraDataEmpty() {
+    require contractAddressesLinked();
+    env e; calldataarg args; env e2; env e3;
+
+    submitReportData(e,args);
+    submitReportExtraDataEmpty(e2);
```

Comment:
Correct me If I'm wrong, but this rule can be generalized for any method in-between: no matter what happens between the two submissions, one cannot submit the same report twice (since it represents the same ref slot).

Reply:
Yes, you are correct. I am adding a parametric rule to check the "any method called" case, which is more general than the current rule. Thanks.